### ISSUE TITLE
Json Editor and edit mode

### ISSUE THREAD

**Issue by kfk** [jupyterlab/jupyterlab#7131]
It seems at this point there is no way to edit a json file in a tree structure, we can see json in a tree structure only in view mode.

Is there a plan to be able to edit json in a more user friendly way? Or is there a way to do it now?

Related issues
https://github.com/jupyterlab/jupyterlab/issues/6185

**Comment by jasongrout** [comment]
The JSON editor is view only. I don't know anyone working on an editor, but that would be great to have! Help is welcome!

**Comment by bbennett36** [comment]
Couldn't we just treat this as a text file for editing?

**Comment by ian-r-rose** [comment]
@bbennett36 Yes, you can already open the JSON file with the normal text editor by right-clicking on it in the file browser and opening it with the Editor. I believe @jasongrout was suggesting an editable JSON tree view, which does not exist at the moment.

**Comment by bbennett36** [comment]
@ian-r-rose ahhhh. I didn't look in the "open with" tab. Thanks!

**Comment by stefaneidelloth** [comment]
As a work around, I created an edit function based on py-jsoneditor:

`pip install jsoneditor`

```
import jsoneditor
import threading
import functools

def edit(json_data):

    result = None

    def data_changed(threading_event, json):
        nonlocal result
        result = json
        threading_event.set()

    event = threading.Event()
    callback_with_event = functools.partial(data_changed, event)
    jsoneditor.editjson(json_data, callback_with_event)
    event.wait()

    if result is not None:
        return result
    else:
        return json_data
```

Instead of opening a new tab I would prefer to edit the json data in some ipywidget or popup window. 
If there is a solution for that, please let me know.

**Comment by AyushmunotSG18** [comment]
Hey all,
Is there any update on this issue. I am still not able to open json file in edit mode. It is still in view mode onlt
